### PR TITLE
Reactive database events

### PR DESCRIPTION
**Pull request by willsb**
This adds events that will fire whenever a time entry is added or updated. Needed for the UI to remain up-to-date®

Squash whenever.

### CODE REVIEW

File: Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs [hunk 1]
```diff
@@ -105,14 +113,26 @@ public class TheStartMethod : TimeEntryDataSourceTest
...
                 TimeEntriesSource.CurrentlyRunningTimeEntry.Where(te => te != null).Subscribe(observer);
 
                 await TimeEntriesSource.Start(ValidTime, ValidDescription, true);
 
                 var currentlyRunningTimeEntry = observer.Messages.Single().Value.Value;
                 await Repository.Received().Create(Arg.Is<IDatabaseTimeEntry>(te => te .Start == currentlyRunningTimeEntry.Start));
             }
+
+            [Fact]
+            public async ThreadingTask NotifiesATimeEntryCreation()
```
Comment (amulware) [inline Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs]: I find this test name confusing.

File: Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs [hunk 2]
```diff
@@ -178,14 +198,25 @@ public void ThrowsIfThereAreNoRunningTimeEntries()
...
                 var observable = TimeEntriesSource.Stop(ValidTime);
                 observable.Subscribe(observer);
 
                 observer.Messages.Single().Value.Exception.Should().BeOfType<InvalidOperationException>();
             }
-        }

 
+            [Fact]
+            public async ThreadingTask NotifiesATimeEntryUpdate()
```
Comment (amulware) [inline Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs]: ditto

File: Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs (hunk 1 above)
Comment (willsb) [inline Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs]: I request a name you don't find confusing, since I find this to be clear enough

Comment (amulware) [inline Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs]: But... the name does not make sense grammatically, does it? 'this notifies a time entry creation' - what?

File: Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs [hunk 3]
```diff
@@ -134,11 +154,11 @@ public TheStopMethod()
                              .Return(timeEntries)
                              .Select(x => x.Where(callInfo.Arg<Func<IDatabaseTimeEntry, bool>>())));
             }
-             
+
             public async ThreadingTask UpdatesTheTimeEntrySettingItsStopTime()
```
Comment (arturs105) [inline Toggl.Foundation.Tests/DataSources/TimeEntriesDataSourceTests.cs]: This is missing the `FactAttribute`.

**Pull request merged**